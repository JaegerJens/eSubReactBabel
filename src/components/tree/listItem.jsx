import React from "react";

export const ListItem = ({number, value, onClick, isSelected}) => {

    const onClickHandler = () => onClick(number, value);

    if (isSelected) {

        return <li className="selected" onClick={onClickHandler}>{value}</li>;

    }

    return <li onClick={onClickHandler}>{value}</li>;

};

export default ListItem;
