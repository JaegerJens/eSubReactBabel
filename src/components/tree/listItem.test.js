import React from "React";
import listItem from "./listItem.jsx";
import renderer from "react-test-renderer";

const emptyClickHandler = () => true;

describe("ListItem component", () => {

    it("matches selected snapshot", () => {

        const props = {
            "isSelected": true,
            "number": 3,
            "onClick": emptyClickHandler,
            "value": "id5424"
        };

        expect(listItem(props)).toMatchSnapshot();

    });
    it("matches unselected snapshot", () => {

        const props = {
            "isSelected": false,
            "number": 4,
            "onClick": emptyClickHandler,
            "value": "id3424"
        };

        expect(listItem(props)).toMatchSnapshot();

    });

});
