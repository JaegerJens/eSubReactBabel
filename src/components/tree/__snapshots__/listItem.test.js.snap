// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component matches selected snapshot 1`] = `
<li
  className="selected"
  onClick={[Function]}
>
  id5424
</li>
`;

exports[`ListItem component matches unselected snapshot 1`] = `
<li
  onClick={[Function]}
>
  id3424
</li>
`;
